include/seqan3/search/algorithm/search.hpp

```diff
@@ -55,6 +82,10 @@ template <FmIndex index_t, typename queries_t, typename configuration_t>
 //!\endcond
 inline auto search(queries_t && queries, index_t const & index, configuration_t const & cfg)
 {
+    static_assert(std::ranges::SizedRange<queries_t>, "The query sequence/range must model SizedRange.");
+    static_assert(dimension_v<queries_t> == 1u || std::ranges::SizedRange<value_type_t<queries_t>>,
+                  "The elements of the range of queries must model SizedRange.");
+
```

Comment:
If the stuff required in the requires clause is also just for security and not for overload resolution you might want to directly change this if you are touching this code anyway